File: backend/src/main/java/com/xxx/animal/entity/Knowledge.java
package com.xxx.animal.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

@Data
@TableName("knowledge")
public class Knowledge {
    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String content;
    private Integer sortOrder;
}

File: frontend/src/api/knowledge.ts
import { http } from './http'

export interface KnowledgeItem {
  id: number
  title: string
  content: string
  sortOrder: number
}

export async function getKnowledgeList() {
  const { data } = await http.get('/api/knowledge')
  return data as { code: number; message: string; data: KnowledgeItem[] }
}

export async function getKnowledgeDetail(id: number | string) {
  const { data } = await http.get(`/api/knowledge/${id}`)
  return data as { code: number; message: string; data: KnowledgeItem | null }
}
